src/main/java/com/startstepszalando/ecommerceshop/product/service/ProductService.java

```diff
@@ -37,7 +46,53 @@ public Page<Product> getAllProducts(int page, int size) throws ProductNotFoundEx
         return products;
     }
 
+    public Product createProduct(Product product, Long adminId) throws
+            DuplicateProductException, InsufficientStockException {
+        if (productRepository.hasEntityWithNameAndPrice(product.getName(), product.getPrice())) {
+            throw new DuplicateProductException(String.format("Product with name %s and price %.2f already exists", product.getName(), product.getPrice()));
+        }
+
+        User admin = userService.findById(adminId)
+                .orElseThrow(() -> new UserNotFoundException("Invalid ID: user not found"));
+
+        if (!isAdminUser(admin)) {
```

Comment:
I would rather prefer to show someone who sees my api response as forbidden than letting them know that they have got an admin id.